includes/class.ParserController.inc.php

```diff
@@ -1725,23 +1725,29 @@ public function setup_directories()
 		/*
 		 * If we cannot write files to the downloads directory, then we can't export anything.
 		 */
-		if (is_writable($this->downloads_dir) === FALSE)
+		try
 		{
-			$this->logger->message('Error: ' . $this->downloads_dir . ' could not be written to, so bulk
-				download files could not be exported', 10);
-			return FALSE;
-		}
+			foreach (array('code-json', 'code-text', 'code-xml', 'images') as $data_dir)
+			{
 
-		foreach (array('code-json', 'code-text', 'code-xml', 'images') as $data_dir)
-		{
+				$this->logger->message('Creating "' . $this->downloads_dir . $data_dir . '"', 4);
 
-			$this->logger->message('Creating "' . $this->downloads_dir . $data_dir . '"', 4);
+				/*
+				* If the JSON directory doesn't exist, create it.
+				*/
+				$this->mkdir($this->downloads_dir . $data_dir);
 
-			/*
-			 * If the JSON directory doesn't exist, create it.
-			 */
-			$this->mkdir($this->downloads_dir . $data_dir);
+			}
 
+			$this->logger->message('Created output directories for bulk download files', 5);
+		}
+		catch (Exception $e)
+		{
+			$this->logger->message('Error: ' . $this->downloads_dir . ' could not be written to, so bulk
+				download files could not be exported', 10);
+			$this->rmdir($this->downloads_dir);
```

Comment:
This is problematic - just because you can't create a new directory doesn't mean you want to blow away all the old data in the parent directory that's there.  Remove this line and I think we're :+1: on this otherwise.